code/modules/client/asset_cache.dm

```diff
@@ -397,16 +397,17 @@ GLOBAL_LIST_EMPTY(asset_datums)
 			if (machine)
 				item = machine
 		var/icon_file = initial(item.icon)
+		var/all_states = icon_states(icon_file)
 		var/icon/I = icon(icon_file, initial(item.icon_state), SOUTH)
 
 		// computers (and snowflakes) get their screen and keyboard sprites
 		if (ispath(item, /obj/machinery/computer) || ispath(item, /obj/machinery/power/solar_control))
 			var/obj/machinery/computer/C = item
 			var/screen = initial(C.icon_screen)
 			var/keyboard = initial(C.icon_keyboard)
-			if (screen)
+			if (screen && screen in all_states)
 				I.Blend(icon(icon_file, screen, SOUTH), ICON_OVERLAY)
-			if (keyboard)
+			if (keyboard && keyboard in all_states)

```

Comment:
more parenthesis, this doesn't do what you think it does

Reply:
of course